kong/clustering.lua

```diff
@@ -627,6 +657,17 @@ end
 function _M.init_worker(conf)
   assert(conf, "conf can not be nil", 2)
 
+  local plguins = assert(kong.db.plugins:get_handlers())
+  table.sort(plguins, function(a, b)
+    return a.name:lower() < b.name:lower()
+  end)
+
+  local sha256 = require("resty.sha256"):new()
+  for _, p in ipairs(plguins) do
+    assert(sha256:update(p.name .. (p.handler.VERSION or "0.0.0")))
```

Comment:
I have seen many times that we forgot to bump version or have bumped it wrong according to semantic versioning, and it might be even more common with external plugins.

Also it seems like we are combining everything in same hash. I am not sure how we can find the output of such hash telling us whether the Kong is compatible or not. It just says that this is the hash of plugins, which may contain external plugins too? Is it just to check if data planes have same hash or different?

I am wondering if hash is good enough, what about just reporting things as JSON:

```
{
  "<plugin-name>": "<plugin-version>",
  ...
}
```

We can make the hash out of the whole thing, and store it to `version_hash` or something, but I think it might be beneficial to see what the versions really are, and not just some hash that does not have any meaning other than being random string.

Also `"0.0.0"`  feels like an unnecessary magic value, `nil`/`null`/ `"unknown"` would be better.